### DISCUSSION TITLE
Missing "configure" script or prerequisites list

### DISCUSSION THREAD

**Discussion by ferrieux** [martinpaljak/GlobalPlatformPro#372] [Ideas]
The project is awesome, but the very first step, installing the prerequisites, is mysterious, as there is no "configure" script and the provided Makefile doesn't work out of the box (obviously needing a `$JAVA_HOME`, some version of jre, and possibly dozens of other dependencies).

The intended population here is telecom-savvy people who just don't eat and sleep with *java* ;)

One way to help beginners is to automate (with a `./configure` or `make prereq` or similar script).

Another is to document ("please install JRE version >=X and set JAVA_HOME to ...")

**Comment by martinpaljak** [comment]
There's a link to the wiki on the main readme, which clearly states how this should be run. https://github.com/martinpaljak/GlobalPlatformPro/wiki/Getting-Started#java

The makefile exists purely for personal use, to be honest. If you don't do Java development, you're probably after the pre-built .jar file, so the presence of a makefile (or god forbid an autoconf script) is somewhat irrelevant?

**Reply by ferrieux** [reply]
Ah OK. Somehow missed that, from a "git clone ... ; grep -r jdk".
Having part of the documentation in the files, part outside (in the Wiki) is a bit confusing ;)
And yes, the Makefile shouldn't be there at all if it's not supposed to be used...

Regarding the "if I don't do java" part: in *most* open source projects, there is a clear distinction between users and contributors. Most end up with an "INSTALL" section in their `README` which is merely `./configure && make && sudo make install` , which *is* supposed to be run by a user who doesn't have the slightest idea about the underlying language(s) used by the project.

(The reason a non-java-programming user might be here is, for example, to enumerate the applets running on a given card, or to install one developed by another person.)

**Comment by martinpaljak** [comment]
Also, I should probably add this to the FAQ: https://github.com/martinpaljak/GlobalPlatformPro/issues/293

The intended audience is not telecom (in fact working with SIM cards is not a goal and requires a lot more effort than "traditional" smart cards for PKI-alike purposes)
